File: cpp/1925.cpp
class Solution {
public:
  int countTriples(int n) {
    int result = 0;
    for(int a = 2; a * a < n; a++){
      for(int b = 1; b < a; b++){
        if (__gcd(a, b) == 1 && (a - b) % 2 == 1) {
          int c = a * a + b * b;
          if (c > n) break;
          result += 2 * (n / c); 
        }
      }
    }
    return result;
  }
};

File: python/1925.py
from math import gcd

class Solution:
    def countTriples(self, n: int) -> int:
        result = 0
        for a in range(2, n + 1):
            for b in range(1, a):
                if gcd(a, b) == 1 and (a - b) % 2 == 1:
                    c = a * a + b * b
                    if c > n:
                        break
                    result += 2 * (n // c)
        return result